### PR TITLE
distinguish replies from incoming requests

### PR DESCRIPTION
**Pull request by alovak**
The current implementation has the issue: when we send a message with STAN and wait for the response we may receive an incoming request from the server with the same STAN (server may have own stan sequence). In this case, we mistakenly find reply channel using incoming request STAN and send it there instead of the response that will arrive next.

This PR fixes the issue by distinguishing replies from incoming requests using MTI function code (3rd digit).

### CODE REVIEW

File: connection.go [hunk 1]
```diff
@@ -328,6 +328,47 @@ func requestID(message *iso8583.Message) (string, error) {
 	return stan, nil
 }
 
+const (
+	// position three of the MTI specifies the message function which
```
Comment (adamdecaf) [inline connection.go:332]:
```suggestion
	// position of the MTI specifies the message function which
```

File: connection.go [hunk 2]
```diff
@@ -328,6 +328,47 @@ func requestID(message *iso8583.Message) (string, error) {
...
+	// defines how the message should flow within the system.
+	messageFunctionIndex = 2
+
+	// there will be no matching request with such ID
+	// so incoming messages will always be handled by
+	// inbound message hanlder
+	incomingRequestID = "incoming request"
```
Comment (adamdecaf) [inline connection.go:339]: Will this duplicate items in the map?

**Review by adamdecaf — APPROVED** [review]

File: connection.go (hunk 2 above)
Comment (alovak) [reply connection.go:339]: it's for reading, not for writing. it will never match the reply channel.

Comment (alovak) [reply connection.go:339]: oh wait. that's a code from my previous iteration... I have to remove it. Thanks for catching it!

**Pull request merged**